client/src/components/Map/UCIMap.js

```diff
@@ -72,51 +239,56 @@ export default class UCIMap extends PureComponent {
 
         // Tracks courses that have already been pinned on the map, so there are no duplicates
         let pinnedCourses = new Set();
+        let index = 0;
 
-        this.state.eventsInCalendar.forEach((event) => {
-            // Filter out those in a different schedule or those not on a certain day (mon, tue, etc)
-            if (
-                event.isCustomEvent ||
-                !event.scheduleIndices.includes(this.state.currentScheduleIndex) ||
-                !event.start.toString().includes(DAYS[this.state.day])
+        // Filter out those in a different schedule or those not on a certain day (mon, tue, etc)
+        this.state.eventsInCalendar
+            .filter(
+                (event) =>
+                    !(
+                        event.isCustomEvent ||
+                        !event.scheduleIndices.includes(this.state.currentScheduleIndex) ||
+                        !event.start.toString().includes(DAYS[this.state.day])
+                    )
             )
-                return;
+            .sort((event, event2) => event.start - event2.start)
+            .forEach((event) => {
+                // Get building code, get id of building code, which will get us the building data from buildingCatalogue
+                const buildingCode = event.bldg.split(' ').slice(0, -1).join(' ');
+                const id = locations[buildingCode];
+                const locationData = buildingCatalogue[id];
+                const courseString = `${event.title} ${event.sectionType} @ ${event.bldg}`;
 
-            // Get building code, get id of building code, which will get us the building data from buildingCatalogue
-            const buildingCode = event.bldg.split(' ')[0];
-            const id = locations[buildingCode];
-            const locationData = buildingCatalogue[id];
-            const courseString = `${event.title} ${event.sectionType} @ ${event.bldg}`;
+                index++; // index temp fix for courses in same building
```

Comment:
Is it actually a "temp" fix? Seems like a solid fix to me
```suggestion
                index++; // always increment index to account for courses within the same building
```